x-pack/plugins/fleet/server/collectors/agent_collectors.ts

```diff
@@ -47,3 +49,103 @@ export const getAgentUsage = async (
     updating,
   };
 };
+
+export interface AgentData {
+  agent_versions: string[];
+  agent_checkin_status: {
+    error: number;
+    degraded: number;
+  };
+  agent_checkin_status_last_1h: {
+    error: number;
+    degraded: number;
+  };
+}
+
+const DEFAULT_AGENT_DATA = {
+  agent_versions: [],
+  agent_checkin_status: { error: 0, degraded: 0 },
+  agent_checkin_status_last_1h: { error: 0, degraded: 0 },
+};
+
+export const getAgentData = async (
+  esClient: ElasticsearchClient,
+  abortController: AbortController
+): Promise<AgentData> => {
+  try {
+    const transformLastCheckinStatusBuckets = (resp: any) =>
+      ((resp?.aggregations?.last_checkin_status as any).buckets ?? []).reduce(
+        (acc: any, bucket: any) => {
+          if (acc[bucket.key] !== undefined) acc[bucket.key] = bucket.doc_count;
+          return acc;
+        },
+        { error: 0, degraded: 0 }
+      );
+    const response = await esClient.search(
+      {
+        index: AGENTS_INDEX,
+        size: 0,
+        aggs: {
```

Comment:
It would be helpful to know how many active agents are assigned to a policy as well. There have been questions in the past about the max num of agents a user would reasonably actually enroll in a single policy. Might be easy to add an additional terms agg here for the top 10 `policy_id`

Reply:
added